napari/layers/surface/surface.py

```diff
@@ -282,14 +289,37 @@ def gamma(self, value):
     def _set_view_slice(self):
         """Sets the view given the indices to slice with."""
 
-        not_disp = list(self.dims.not_displayed)
-        disp = list(self.dims.displayed)
-        indices = np.array(self.dims.indices)
+        # Take vertex_values dimensionality into account if more than one value
+        # is provided per vertex.
+        if self.vertex_values.ndim > 1:
+            values = self.vertex_values[
+                self.dims.indices[: -self.vertices.shape[1]]
+            ]
+            if values.ndim > 1:
+                # If not all dimensions corresponding to the vertex_values
+                # are being sliced through, average over any remaining ones
+                # so that there is always only value per vertex.
+                values = values.mean(axis=tuple(range(0, values.ndim - 1)))
```

Comment:
This seems very brittle and undocumented. Should we not raise an error in this case?

Reply:
hmm, i'd prefer not to error here, but agree it's a bit strange. Can we leave for now?

Reply:
non-breaking, non-documented, deep-in-the-code behaviour tends to never be "for now". Unless we can come up with a reasonable default behaviour here, in my opinion, this should be an error: you are trying to display multiple values as a single one. For all you know, these could be measurements in different units! Without a clear use example, it just seems wrong to do this. An error with a *clear* message of how users could view their data would be perfectly acceptable imho: it's a simple `.mean()` call outside of napari, and outside napari is where it should be: we shouldn't do statistical processing for users.

What I *might* live with is a *warning* that the data is being munged and, maybe, grabbing only the 0th element among the remaining axes.

But my preference is still to error out. I don't see how anything else would be correct in the majority of cases. And, actually, how is it even possible to land in this bit of code?

Reply:
Ok, these are good points, and I am changing the current behavior. You can land in the code by changing which dimensions are displayed vs non-displayed (i.e. by clicking the roll axes button in the nilearn example).

Raising an error wasn't so great as the old data from the layer would stay on the screen, I now emit a warning and make all the data disappear, which I think is the right call as it doesn't really make sense to even grab the `0th` axis or something in this case. It really means nothing should be shown.